### PR TITLE
Show new pages on update

### PR DESCRIPTION
**Pull request by cassidyjames**
Based on something @danrabbit had mentioned. Basically, this would allow us to use onboarding to introduce new major features for existing installs by tracking which pages have been seen.

- [x] Track which views have been viewed
- [x] …even when using the pagers
- [x] Skip already-viewed views
- [x] Create a new "udpate" view to show after first-run
- [x] Don't always quit after first-run
- [x] Quit if there are no feature views to show
- [x] Mark children as viewed when skipping all
- [ ] …?

### CODE REVIEW

File: src/MainWindow.vala [hunk 1]
```diff
@@ -129,5 +155,14 @@ public class Onboarding.MainWindow : Gtk.Window {
             stack.visible_child_name = "finish";
         });
     }
-}
 
+    private void mark_viewed (string view_name) {
+        if (!(name in viewed)) {
```
Comment (danrabbit) [inline src/MainWindow.vala]:
```suggestion
        if (!(view_name in viewed)) {
```

File: src/MainWindow.vala [hunk 2]
```diff
@@ -56,6 +66,24 @@ public class Onboarding.MainWindow : Gtk.Window {
...
+
+            if (view_name in viewed) {
+                view.destroy ();
+            }
+        }
+
+        // Bail if there are no feature views
+        if (stack.get_children ().length () < 2) {
+            // FIXME: Okay, this isn't quitting…
+            Onboarding.App.instance.quit ();
```
Comment (danrabbit) [inline src/MainWindow.vala]: We don't need to have the application keep track of its instance. GLib does that for us

```suggestion
            GLib.Application.get_default ().quit ();
```

**Comment by danrabbit** [comment]
I think, like you said in slack, that we need to add a special behavior for the skip all button to just add everything to "viewed"

File: src/MainWindow.vala [hunk 3]
```diff
@@ -28,14 +31,24 @@ public class Onboarding.MainWindow : Gtk.Window {
     }
 
     construct {
+        settings = new GLib.Settings ("io.elementary.onboarding");
```
Comment (danrabbit) [inline src/MainWindow.vala]: If this is static, you probably want to assign it inside a `static construct {}`

**Comment by danrabbit** [comment]
Oh also, remember to add updates view to POTFILES

**Comment by cassidyjames** [comment]
@danrabbit okay maybe it's moot, but should we call "viewed" something else since "skip all" marks them as "viewed" even thought they've not actually been seen?

File: src/MainWindow.vala [hunk 4]
```diff
@@ -115,19 +149,36 @@ public class Onboarding.MainWindow : Gtk.Window {
...
+            if (index < current_views.length () - 1) {
+                stack.visible_child = current_views.nth_data (index + 1);
             } else {
-                Onboarding.App.settings.set_boolean ("first-run", false);
                 destroy ();
             }
         });
 
         skip_button.clicked.connect (() => {
+            foreach (Gtk.Widget view in views) {
```
Comment (danrabbit) [inline src/MainWindow.vala]: indentation?

**Comment by danrabbit** [comment]
@cassidyjames I dunno I think it's fine even if it's not 100% always exactly perfectly true

**Pull request merged**